test/12_conditional_commit.ts

```diff
@@ -173,41 +178,287 @@ describe('Create Voucher sets and commit to vouchers with token conditional comm
   }
 
   describe('TOKEN SUPPLY CREATION WITH TOKEN CONDITIONAL COMMIT (Create Voucher Set)', () => {
-    let timestamp;
-    describe('TKNTKN', () => {
-      async function generateInputs(
-        account: Account,
-        deposit: number | string,
-        qty: number | string
-      ) {
-        const txValue = BN(deposit).mul(BN(qty));
+    const paymentMethods = {
+      ETHETH: 1,
+      ETHTKN: 2,
+      TKNETH: 3,
+      TKNTKN: 4,
+    };
+    let timestamp, tokenSupplyKey: BigNumber, promiseId: string;
+
+    async function generateInputs(
+      account: Account,
+      deposit: number | string,
+      qty: number | string
+    ) {
+      const txValue = BN(deposit).mul(BN(qty));
+
+      const nonce = await contractBSNTokenDeposit.nonces(account.address);
+
+      const digest = await getApprovalDigest(
+        contractBSNTokenDeposit,
+        account.address,
+        contractBosonRouter.address,
+        txValue,
+        nonce,
+        deadline
+      );
+
+      const {v, r, s} = ecsign(
+        Buffer.from(digest.slice(2), 'hex'),
+        Buffer.from(account.privateKey.slice(2), 'hex')
+      );
+
+      return {txValue, v, r, s};
+    }
+
+    before(async () => {
+      timestamp = await Utils.getCurrTimestamp();
+      constants.PROMISE_VALID_FROM = timestamp;
+      constants.PROMISE_VALID_TO = timestamp + 2 * constants.SECONDS_IN_DAY;
+
+      tokenSupplyKey = calculateTokenSupplyKey(constants.ONE);
+      promiseId = keccak256(
+        solidityPack(
+          ['address', 'uint256', 'uint256', 'uint256'],
+          [
+            users.seller.address,
+            constants.ZERO,
+            constants.PROMISE_VALID_FROM,
+            constants.PROMISE_VALID_TO,
+          ]
+        )
+      );
+    });
 
-        const nonce = await contractBSNTokenDeposit.nonces(account.address);
+    describe('ETHETH', () => {
+      beforeEach(async () => {
+        await deployContracts();
+      });
 
-        const digest = await getApprovalDigest(
-          contractBSNTokenDeposit,
-          account.address,
-          contractBosonRouter.address,
-          txValue,
-          nonce,
-          deadline
+      it('Should be able to create Voucher with gate address', async () => {
+        const txValue = BN(constants.PROMISE_DEPOSITSE1).mul(
+          BN(constants.QTY_10)
         );
+        expect(
+          await contractBosonRouter
+            .connect(users.seller.signer)
+            .requestCreateOrderETHETHConditional(
+              [
+                constants.PROMISE_VALID_FROM,
+                constants.PROMISE_VALID_TO,
+                constants.PROMISE_PRICE1,
+                constants.PROMISE_DEPOSITSE1,
+                constants.PROMISE_DEPOSITBU1,
+                constants.QTY_10,
+              ],
+              contractGate.address,
+              '0',
```

Comment:
This is a small thing, but I think it would be helpful to application developers who need to call `requestCreateOrderETHETHConditional` to be able to use the tests as a good example of how to call the conditional commit functions. Passing in `0` could be misleading. Can you make this a number so it's more representational? 
The same goes for the other `*Conditional` functions.

Reply:
Introduced new constant `EMPTY_NFT_TOKEN_ID` for these cases. Resolved with next commit.